src/main/java/seedu/address/commons/util/DateTimeUtil.java

```diff
@@ -1,11 +1,14 @@
 package seedu.address.commons.util;
 
 import java.time.format.DateTimeFormatter;
+import java.time.format.ResolverStyle;
 
 /**
  * A class for formatting and parsing LocalDate objects
  */
 public class DateTimeUtil {
-    public static final DateTimeFormatter DEFAULT_DATE_PARSER = DateTimeFormatter.ofPattern("yyyy-MM-dd");
-    public static final DateTimeFormatter DEFAULT_DATE_FORMATTER = DateTimeFormatter.ofPattern("dd-MMM-yyyy");
+    public static final DateTimeFormatter DEFAULT_DATE_PARSER = DateTimeFormatter.ofPattern("uuuu-MM-dd")
+            .withResolverStyle(ResolverStyle.STRICT);
+    public static final DateTimeFormatter DEFAULT_DATE_FORMATTER = DateTimeFormatter.ofPattern("dd-MMM-uuuu")
+            .withResolverStyle(ResolverStyle.STRICT);
```

Comment:
What are the differences between `yyyy` and `uuuu`?

Reply:
To prevent the parser from rounding off invalid dates, need to use ResolverStyle.Strict, which needs to use `uuuu` if you do not want to specify the era. I was referring to this which says that if `yyyy` is used should specify the era. 
https://stackoverflow.com/questions/41177442/uuuu-versus-yyyy-in-datetimeformatter-formatting-pattern-codes-in-java

Reply:
If we use this then negative years are valid? 0000 = 1 BC, -0001 = 2 BC?

Reply:
I think our formatter doesn't look very nice when it displays a negative year

Reply:
That's true, but I think that's a minor issue since I think it's considered extreme user input to input a negative year? If you have any ideas on how to enforce the strict requirement without changing `yyyy` to `uuuu` can let me know, but I think in general my understanding is that `uuuu` is safer than `yyyy`

Reply:
@darkmoongreatsword do you have any opinions on this?

Reply:
Ok I think better to use `uuuu`.